src/configAPIClient.ts

```diff
@@ -1,13 +1,37 @@
-export default function ConfigAPIClient() {
-    this.getSDKConfiguration = function(
+import { DataPlanConfig } from '@mparticle/web-sdk';
+import { KitConfigs, MParticleWebSDK, SDKInitConfig } from './sdkRuntimeModels';
+
+export type SDKInitFunction = (
+    apiKey: string,
+    config: SDKInitConfig,
+    mpInstance: MParticleWebSDK
+) => void;
+
+export interface IConfigResponse {
+    appName: string;
+    kitConfigs: KitConfigs[];
+}
+
+export interface IConfigAPIClient {
+    getSDKConfiguration: (
+        apiKey: string,
+        config: SDKInitConfig,
+        completeSDKInitialization: SDKInitFunction,
+        mpInstance: MParticleWebSDK
+    ) => void;
+}
+
+export default function ConfigAPIClient(this: IConfigAPIClient) {
```

Comment:
As part of this migration this should be converted into a TS class instead of remaining an ES5 class-like function:

```
export class ConfigAPIClient {
   constructor(blah blah)
}
```

Reply:
I would normally agree, but I want to focus this migration on a direct translation to typescript and fleshing out type definitions rather than making changes to core functionality or converting functions to classes to avoid any potential side effects.

Also, as we are using similar API clients elsewhere in the SDK, I would rather combine these clients into a single API client class as part of a future pull request.

Reply:
SGTM